src/solvers/flattening/boolbv_get.cpp

```diff
@@ -215,6 +217,17 @@ exprt boolbvt::bv_get_rec(const exprt &expr, const bvt &bv, std::size_t offset)
     {
       return constant_exprt{irep_idt(), type};
     }
+    else if(type.id() == ID_enumeration)
+    {
+      auto &elements = to_enumeration_type(type).elements();
+      mp_integer int_value = binary2integer(value, false);
+      irep_idt element;
+      if(int_value >= elements.size())
+        element = irep_idt{};
```

Comment:
❓ Where will we fall over when encountering this?

Reply:
Returning nil instead.